app/services/resolve_account_service.rb

```diff
@@ -82,7 +82,7 @@ def process_webfinger!(uri, redirected = false)
     @webfinger                           = webfinger!("acct:#{uri}")
     confirmed_username, confirmed_domain = @webfinger.subject.gsub(/\Aacct:/, '').split('@')
 
-    if confirmed_username.casecmp(@username).zero? && confirmed_domain.casecmp(@domain).zero?
+    if (confirmed_username.casecmp(@username).zero? && confirmed_domain.casecmp(@domain).zero?) || @webfinger.aliases.any? { |s| s.casecmp("acct:#{@username}@#{@domain}").zero? }
```

Comment:
Wait, no, don't do that. It would be ok for e.g. an account name change, but here, an attacker could use a subject with a totally different `domain` that it isn't authoritative for, and your check would just accept it as long as the original queried `acct:` URI is in the aliases. I don't think you can cut on that extra query if the domains are different.